File: src/day1/Day1.java
package adventjava;

import java.io.BufferedReader;
import java.io.File;
import java.io.FileReader;
import java.io.IOException;

public class Day1 {
    public static void main(String[] args) {
        partOne();
    }

    public static void partOne() {
        try {
            File file = new File("./src/day1input.txt");
            FileReader fr = new FileReader(file);
            BufferedReader br = new BufferedReader(fr);
            int answer = br.lines()
                    .map((x) -> Integer.parseInt(x))
                    .reduce(0, (first, second) -> first < second ? 0 : 1); // does not work, need to find a way to keep a running total
            System.out.println("Part 1: " + answer);
            br.close();
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}

File: src/day1/day1.py
import math

def partOne():
    file = open("../day1input.txt", "r")
    previousNumber = math.inf
    currentNumber = 0
    count = 0
    
    for line in file:
        currentNumber = int(line)
        if (currentNumber > previousNumber):
            count = count + 1
        previousNumber = currentNumber
    
    print(count)
    file.close()
    
def partTwo():
    file = open("../day1input.txt", "r")
    first = int(file.readline())
    second = int(file.readline())
    windowOne = math.inf
    windowTwo = 0
    count = 0
    
    for line in file:
        third = int(line)
        windowTwo = first + second + third
        if (windowTwo > windowOne):
            count = count + 1
        first = second
        second = third
        windowOne = windowTwo
        
    print(count)
    file.close()